zebra-chain/src/serialization/date_time.rs

```diff
@@ -101,6 +133,38 @@ impl Duration32 {
     pub fn to_std(self) -> std::time::Duration {
         self.into()
     }
+
+    /// Returns the time that is `duration` seconds after this time.
+    /// If the calculation overflows, returns `None`.
+    pub fn checked_add(&self, duration: Duration32) -> Option<Duration32> {
+        self.seconds
+            .checked_add(duration.seconds)
+            .map(|seconds| Duration32 { seconds })
```

Comment:
Just thought of an alternative way to implement these `checked_` calculations:

```suggestion
        Some(Duration32 {
            seconds: self.seconds.checked_add(duration.seconds)?,
        })
```

Not really a suggestion, just wanted to mention it in case it's useful :laughing:

Reply:
I think the map is clearer, but it's such a short piece of code the difference is minor.